### DISCUSSION TITLE
Tidepool Authentication API Deprecation

### DISCUSSION THREAD

**Discussion by toddkazakov** [NightscoutFoundation/xDrip#2596] [General]
Hello, I am the lead backend developer at Tidepool. In December 2022 we replaced our legacy authentication system with  Keycloak. As a result, the existing API endpoints are deprecated, but we noticed that they are heavily used xDrip. As we are trying to completely retire the old system, all applications should migrate to the new OpenID Connect compatible implementation.

We don't have timeframe for the removal of the old APIs, but the way xDrip currently utilizes the APIs puts a strain on the new system (existing APIs act as proxy), because it starts a new auth session every time it requires a fresh token, instead of refreshing the session token directly.

I'll be more than happy to assist with setting up an integration/production environment client for the application. Any OIDC compatible library can be used (e.g. https://github.com/openid/AppAuth-Android). Let me know what's the best way to proceed.

**Comment by Navid200** [comment]
@toddkazakov Thanks for reaching out.

Would you mind clicking on "Create issue from discussion" in the sidebar?
Would you please tell me if that option is not available to you?
In that case, would you please open an issue?  But, I would appreciate it if you told me that you don't see that option.

Thanks again

**Reply by toddkazakov** [reply]
@Navid200 done

**Reply by Navid200** [reply]
@toddkazakov 
When you can, please respond to the question in the issue: 
https://github.com/NightscoutFoundation/xDrip/issues/2597#issuecomment-1550997142
